src/assets/CollectibleDetectionController.ts

```diff
@@ -309,15 +314,22 @@ export class CollectibleDetectionController extends BaseController<
     if (!this.isMainnet() || this.disabled) {
       return;
     }
-    const requestedSelectedAddress = this.config.selectedAddress;
+    const { selectedAddress } = this.config;
+
+    let { chainId } = this.config;
+    if (typeof chainId === 'string' && isHexString(chainId)) {
+      chainId = `${parseInt(chainId, 16)}` as `${number}`;
+    } else if (typeof chainId === 'number') {
+      chainId = `${chainId}` as `${number}`;
```

Comment:
Using `as const` here is enough to get TypeScript to accept these values as template literal types. Const assertions are safer than casting directly to the expected type, because it still lets `tsc` type check this for us.
```suggestion
      chainId = `${parseInt(chainId, 16)}` as const;
    } else if (typeof chainId === 'number') {
      chainId = `${chainId}` as const;
```

Reply:
This is what I needed and I did not know it, thanks! I will commit it directly from github.